sympy/stats/stochastic_process_types.py

```diff
@@ -625,9 +680,20 @@ class DiscreteMarkovChain(DiscreteTimeStochasticProcess, MarkovProcess):
 
     def __new__(cls, sym, state_space=S.Reals, trans_probs=None):
         sym = _symbol_converter(sym)
-        state_space = _set_converter(state_space)
-        if trans_probs != None:
+
+        # maybe make trans_probs a MatrixSymbol if it is None
+        if trans_probs is not None:
             trans_probs = _matrix_checks(trans_probs)
+
+        if (state_space is S.Reals) and (trans_probs is not None):
+            # handle symbolic-sized matrix
+            n = trans_probs.shape[0]
+            if isinstance(n, Symbol):
+                state_space = Range(0, n)
+            else:
+                state_space = FiniteSet(*Range(0, n))
```

Comment:
Normal Python `range(n)` could be used here.

Alternatively `Range` represents a sort of `Set` itself so is it possible to use `Range` directly here rather than `FiniteSet`?

Reply:
I think I'll sort this out when dealing with point number 3. The state space for any `RandomVariable` is a `Set` but in this case we need to preserve order. So I'm a little worried with some probability statements that assume it is a set. Hopefully it all works out.